packages/docusaurus-theme-classic/src/remark/admonitions/index.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import visit from 'unist-util-visit';
+import type {Transformer} from 'unified';
+import type {Literal} from 'mdast';
+
+const NEWLINE = '\n';
+
+const config = {
+  tag: ':::',
+  keywords: [
+    'secondary',
+    'info',
+    'success',
+    'danger',
+    'note',
+    'tip',
+    'warning',
+    'important',
+    'caution',
+  ],
+};
+
+function escapeRegExp(s: string): string {
+  return s.replace(/[-[\]{}()*+?.\\^$|/]/g, '\\$&');
+}
+
+export default function plugin(this: any): Transformer {
+  const keywords = Object.values(config.keywords).map(escapeRegExp).join('|');
+  const tag = escapeRegExp(config.tag);
+  const regex = new RegExp(`${tag}(${keywords})(?: *(.*))?\n`);
+  const escapeTag = new RegExp(escapeRegExp(`\\${config.tag}`), 'g');
+
+  // the tokenizer is called on blocks to determine if there is an admonition
+  // present and create tags for it
+  function blockTokenizer(this: any, eat: any, value: string, silent: boolean) {
+    // stop if no match or match does not start at beginning of line
+    const match = regex.exec(value);
+    if (!match || match.index !== 0) {
+      return false;
+    }
+    // if silent return the match
+    if (silent) {
+      return true;
+    }
+
+    const now = eat.now();
+    const [opening, keyword, title] = match;
+    const food = [];
+    const content = [];
+
+    let newValue = value;
+    // consume lines until a closing tag
+    let idx = newValue.indexOf(NEWLINE);
+    while (idx !== -1) {
+      // grab this line and eat it
+      const next = newValue.indexOf(NEWLINE, idx + 1);
+      const line =
+        next !== -1 ? newValue.slice(idx + 1, next) : newValue.slice(idx + 1);
+      food.push(line);
+      newValue = newValue.slice(idx + 1);
+      // the closing tag is NOT part of the content
+      if (line.startsWith(config.tag)) {
+        break;
+      }
+      content.push(line);
+      idx = newValue.indexOf(NEWLINE);
+    }
+
+    // consume the processed tag and replace escape sequences
+    const contentString = content.join(NEWLINE).replace(escapeTag, config.tag);
+    const add = eat(opening + food.join(NEWLINE));
+
+    // parse the content in block mode
+    const exit = this.enterBlock();
+    const contentNodes = this.tokenizeBlock(contentString, now);
+    exit();
+
+    const element = {
+      type: 'admonitionHTML',
+      data: {
+        hName: 'admonition',
```

Comment:
I think this is a pretty good step. This also means users can swizzle `MDXComponents` and re-map `admonition` to some other component, not necessarily `@theme/Admonition`.

Reply:
👍 totally in favor of this